New header style

harshvitra opened the pull request:
Fixes #1381

Some tasks to do :-)
- [x] fix iOS problem
- [x] fix eslint
- [x] fix conflicts
- [x] https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1792#pullrequestreview-350264388
- [ ] donate button for web client showing single project from share link https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1792#issuecomment-598349859

norbertschuler commented:
Is this header really meant to show no back arrow any more, so you can only go back by swiping on iOS (tested with iOS13.3 and iPhone X simulator)? I don't think iOS users like that.

![Simulator Screen Shot - iPhone 11 - 2020-01-10 at 14 47 47](https://user-images.githubusercontent.com/1532418/72157461-4dc55b00-33b8-11ea-839a-ff4d30427bcd.png)
![Simulator Screen Shot - iPhone 11 - 2020-01-10 at 14 46 55](https://user-images.githubusercontent.com/1532418/72157462-4dc55b00-33b8-11ea-94cf-772d3446da0d.png)

norbertschuler commented:
I once more replaced the cross icons with font icons and added a shadow to the white icon.

norbertschuler commented:
@harshvitra I have removed app/assets/images/header/crossBlack.png and app/assets/images/header/crossWhite.png intentionally and replaced them with a font icon. Please let's delete them again, otherwise they are unused in our repository, aren't they?

harshvitra commented:
> @harshvitra I have removed app/assets/images/header/crossBlack.png and app/assets/images/header/crossWhite.png intentionally and replaced them with a font icon. Please let's delete them again, otherwise they are unused in our repository, aren't they?

Sorry I forgot to explain this point - 
Actually in one of our discussion @sagararyal said he wanted rounded ones, so I took it from the zeplin file. We can use either, final call is with Sagar

norbertschuler commented:
> Actually in one of our discussion @sagararyal said he wanted rounded ones, so I took it from the zeplin file. We can use either, final call is with Sagar

O.k., sorry for changing this then. Also I just mentioned this issue in #1859 as there we implemented a complete different solution for a (x) closing button.

norbertschuler commented:
@harshvitra Do we still want to have this PR any more? I think there is a change request pending. Also please merge the develop branch and solve the conflicts.

harshvitra commented:
> @harshvitra Do we still want to have this PR any more? I think there is a change request pending. Also please merge the develop branch and solve the conflicts.

Yes I need this now, right time to work on this PR. 
I have worked on headers for other pages, now its turn for this one

harshvitra commented:
> Please do not leave the white area in the top.
> 
> Example of how it can be solved.
> 
> <img alt="Screen Shot 2020-01-29 at 6 51 29 PM" width="924" src="https://user-images.githubusercontent.com/5401575/73377892-727e6580-42c8-11ea-804c-5e1ba76ab239.png">

This issue was only in iOS, now it is fixed

harshvitra commented:
Don't merge, I'll add the share functionality here

norbertschuler commented on .env.sample:
```diff
@@ -0,0 +1 @@
+googleMapApiKey=
```
This file is deprecated and not used any more. Please delete it.

norbertschuler replied:
Removed the file myself.

norbertschuler commented on app/actions/loadTposAction.js:
```diff
@@ -58,6 +59,7 @@ export function loadProject(plantProject, options = {}) {
       request
         .then(res => {
           debug('========================', res.data);
+          dispatch(setSelectedPlantProjectId(parseInt(plantProject.id)));
```
I assume this could make some of the `selectPlantProjectAction()` calls redundant, but better safe then sorry,

norbertschuler replied:
now we know this call is needed as otherwise http://localhost:8080/project/1 does not work.

norbertschuler commented:
> Don't merge, I'll add the share functionality here

@harshvitra Are you sure, you don't want to do that in another pull request and get this one merged?

harshvitra commented:
> I thought a new route also has to be added into the backend, have you or @jmiridis done this?

@jmiridis what is the process for this ?

harshvitra commented:
> > Don't merge, I'll add the share functionality here
> 
> @harshvitra Are you sure, you don't want to do that in another pull request and get this one merged?

At the end both have to be there in v1.3.2

norbertschuler commented on ios/TreecounterApp.xcodeproj/project.pbxproj:
```diff
@@ -13,10 +13,9 @@
 		13B07FBD1A68108700A75B9A /* LaunchScreen.xib in Resources */ = {isa = PBXBuildFile; fileRef = 13B07FB11A68108700A75B9A /* LaunchScreen.xib */; };
```
I don't see why this PR has to change this file. Probably Xcode just changed some hash keys since you rebuild the project, but I think you could reset this file to the old version and does not have to include any changes of it in this PR.

norbertschuler replied:
Fixed it.

norbertschuler commented on app/components/PlantProjects/PlantProjectFull.native.js:
```diff
@@ -130,10 +137,11 @@ class PlantProjectFull extends React.Component {
+            '/' +
+            this.props.plantProject.id
           }
+          appurl={'weplant://project/' + this.props.plantProject.id}
```
I think you have not yet added the new deep link to Android at https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/36578ce963f4e7c795dfbec944c5aa0b91a42b79/android/app/src/main/AndroidManifest.xml#L72 and to iOS in https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/develop/.well-known/apple-app-site-association?

Check out the guide from Android https://developer.android.com/training/app-links/deep-linking and iOS https://developer.apple.com/ios/universal-links/

norbertschuler replied:
Reminder for myself and anyone else. Test the new deeplink like we did it for redeem trees for Android https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1743#issuecomment-575154906 and iOS https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1743#issuecomment-575160775

norbertschuler replied:
- [ ] For **iOS** `xcrun simctl openurl booted weplant://project/1` or `xcrun simctl openurl booted trilliontreecampaign://project/1`  is working, but `xcrun simctl openurl booted https://www.trilliontreecampaign.org/project/1` is not. The later is even throwing a 404 in the web client as the backend has not defined this route yet:

![Simulator Screen Shot - iPhone 11 - 2020-03-12 at 14 06 25](https://user-images.githubusercontent.com/1532418/76524721-d325c400-646a-11ea-970d-6d44424a0a9e.png)

The share function promotes both kind of deep links saying e.g. "Plant trees by helping  at https://test.trilliontreecampaign.org/project/1 or weplant://project/1".

norbertschuler replied:
- [ ] For **Android** `adb shell am start -W -a android.intent.action.VIEW -d "weplant://project/1"` or `adb shell am start -W -a android.intent.action.VIEW -d "trilliontreecampaign://project/1"` is working, but `adb shell am start -W -a android.intent.action.VIEW -d "https://www.trilliontreecampaign.org/project/1"` is not. This also is throwing a 404 error in the browser instead.

harshvitra commented:
@norbertschuler when you check the web version. I have added a new page to show a single project. 

But apparently it does not have a donate button. In fact the web donate works all in one file

norbertschuler commented on app/containers/SelectedPlantProject/index.js:
```diff
@@ -1,74 +1,57 @@
+  };
+
+  return props.selectedProject ? (
+    <PlantProjectSingle
```
Currently calling http://localhost:8080/project/1 in the web client show an empty page. This has to be fixed before we can merge this PR.

harshvitra replied:
Fixed - https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1792/commits/27d1557cfb2ee47a32cd57c4340edf77f42151b5

norbertschuler commented:
> @norbertschuler when you check the web version. I have added a new page to show a single project.
> 
> * [ ]  But apparently it does not have a donate button. In fact the web donate works all in one file

@harshvitra Can you add a donate button to the web version, so we can merge this PR or do we have to rethink how to handle the project sharing url in the web? Do we want to wait for that before we merge this issue of do it in a separate task? @sagararyal what do you think about this?

harshvitra commented:
> @harshvitra Can you add a donate button to the web version, so we can merge this PR or do we have to rethink how to handle the project sharing url in the web? Do we want to wait for that before we merge this issue of do it in a separate task? @sagararyal what do you think about this?

Honestly I tried but failed with 2-3 approaches, maybe just can't see the right path here

norbertschuler commented:
> Honestly I tried but failed with 2-3 approaches, maybe just can't see the right path here

Have you tried using the existing code to directly jump to a project the same way it get opened from the TPO treecounter of somehow else before you started coding own own single project page? (By the way production has an error trying to open projects other than Plant-for-the-Planet e.g. Weforest or Eden from the TPO treecounter.)

sagararyal commented:
Link is missing:

trilliontreecampaign.org/donate-trees/projectid

![IMG_8843877BC50E-1](https://user-images.githubusercontent.com/5401575/77009556-66c62b80-6968-11ea-9f33-d2bb6ab6886e.jpeg)

sagararyal commented:
Images with leaderboard is also missing.

norbertschuler commented:
> Link is missing:
> 
> trilliontreecampaign.org/donate-trees/projectid

This was my fault because of wrongly removed code. I added it again.

norbertschuler commented:
> Images with leaderboard is also missing.

What does this mean? Which images are missing? (I see flags and user images in the Leaderboard in the app.)

sagararyal commented on app/components/PlantProjects/PlantProjectFull.native.js:
```diff
@@ -130,6 +131,15 @@ class PlantProjectFull extends React.Component {
+            '/' +
+            this.props.plantProject.id
+          }
+          appurl={'weplant://project/' + this.props.plantProject.id}
```
Shared URL here should we web URL.
not weplant://

sagararyal replied:
weplant:// would be added in the website as meta so that deep link would be forwarded.
